### PR TITLE
Fix #1592: remove duplication of Command Names

### LINKED ISSUE #1592: Code Tidy: @JsonTypeName on API Command objects duplicates command names
noticed when doing a review we have code like 

```
@JsonTypeName("dropIndex")
public record DropIndexCommand(
```

And we now have the CommandName enum, so we could do `CommandName.DROP_INDEX.getApiName()` 

this ticket is to update the `@JsonTypeName` for all commands: collections, keyspaces, tables etc .

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Removes duplication of Command name Strings (like "alterTable")

**Which issue(s) this PR fixes**:
Fixes #1592

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Initial Proof-of-Concept to show how this could be done, different from the original plan, due to limitations Java Annotation type imposes.

**Comment by tatu-at-datastax** [comment]
@amorton @maheshrajamani Was hoping to give feedback on whether this is worth pursuing or not? Leads to minor duplication (due to limitations of Java annotations) but can avoid more.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/AlterTableCommand.java [hunk 1]
```diff
@@ -7,10 +7,9 @@
 import org.eclipse.microprofile.openapi.annotations.media.Schema;
 
 @Schema(description = "Command that alters the column definition in a table.")
-@JsonTypeName("alterTable")
+@JsonTypeName(CommandName.Names.ALTER_TABLE_NAME)
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/AlterTableCommand.java:10]: After the last merge I think we can set this as `CommandName.ALTER_TABLE.getApiName()`

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandName.java [hunk 2]
```diff
@@ -96,4 +96,8 @@ public static List<CommandName> filterByTarget(CommandTarget target) {
         .filter(commandName -> commandName.getTargets().contains(target))
         .collect(Collectors.toList());
   }
+
+  public interface Names {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/CommandName.java:100]: Don't think we need this interface we can just use the apiName property of the enum.

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/AlterTableCommand.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/AlterTableCommand.java:10]: Unfortunately no -- annotation values must be actual constants, like String constants; cannot use `Constant.method()` :-(
(I tried that, won't compile)

**Review by maheshrajamani — APPROVED** [review]

**Pull request merged**